src/main/java/io/vertx/gsoc2018/qotd/QuoteOfTheDayVerticle.java

```diff
@@ -31,6 +30,13 @@
 
   @Override
   public void start(Future<Void> startFuture) {
+    Future<Void> dbReady = Future.future();
+    Future<Void> httpServerReady = Future.future();
+
+    // complete startFuture only when db and http server started
+    compose(startFuture, dbReady, httpServerReady);
```

Comment:
Check out the Vert.x core documentation, in particular the _sequential composition_ section

Reply:
Done, replaced with `CompositeFuture.all`